server/budgettracker.data/Repositories/BudgetRepository.cs

```diff
@@ -61,5 +59,24 @@ where Ids.Contains(b.Id)
                 throw new RepositoryException("NOT_DELETED:" + errorIds);
             }
         }
+
+        public async Task<List<Budget>> GetRootBudgets(Guid userId)
+        {
+            List<BudgetModel> rootBudgets = await _dbContext.Budgets.Where(b => b.OwnerId == userId).ToListAsync();
```

Comment:
Might as well add another where statement here so you don't have to query it again. I am also open you keeping how you have it just a thought.

Reply:
I have to do it this way. Entity framework can't use null values in where statements so you have to make a query for what you can first, load it into memory, and then evaluate the rest in memory.